build-system/tasks/helpers.js

```diff
@@ -25,7 +25,9 @@ const file = require('gulp-file');
 const fs = require('fs-extra');
 const globby = require('globby');
 const gulp = require('gulp');
+const gulpIf = require('gulp-if');
 const gulpWatch = require('gulp-watch');
+const istanbul = require('gulp-istanbul');
```

Comment:
Why are we using `instanbul` when we have `codecov.io`?

Reply:
My understanding (and @rsimha please correct me if I'm wrong) is that Codecov receives and processes and displays the results of coverage reports, while `istanbul` does the actual instrumentation to generate those reports, replacing Mocha's built in reporting. Is that accurate?

Reply:
Thanks for clarifying. Looks like we use `karma-coverage-istanbul-reporter`.

Reply:
> Codecov receives and processes and displays the results of coverage reports

Correct.

> `istanbul` does the actual instrumentation to generate those reports

Correct.

> replacing Mocha's built in reporting

This is incorrect. Mocha's build-in reporting is rudimentary, but it supports external reporting plugins for test results. This PR is about code coverage for integration tests, which has never worked correctly because it's not straightforward to measure coverage of JS files served by a webserver.